src/main/scala/lc/Main.scala

```diff
@@ -14,37 +24,54 @@ trait ChallengeProvider {
 class Captcha {
   val con: Connection = DriverManager.getConnection("jdbc:h2:./captcha", "sa", "")
   val stmt: Statement = con.createStatement()
-  stmt.execute("CREATE TABLE IF NOT EXISTS challenge(token varchar, id varchar, secret varchar, image blob)")
-  val insertPstmt: PreparedStatement = con.prepareStatement("INSERT INTO challenge(token, id, secret) VALUES (?, ?, ?)")
-  val selectPstmt: PreparedStatement = con.prepareStatement("SELECT secret FROM challenge WHERE token = ?")
-  
-  def getCaptcha(): Boolean = {
-    val provider = new FilterChallenge
-    val (token, image) = this.getChallenge(provider)
-    image.output(new File("Captcha.png"))
-    println(s"Token: ${token}")
-    println("Enter your answer: ")
-    val answer = scala.io.StdIn.readLine()
-    this.getAnswer(token, answer, provider)
+  stmt.execute("CREATE TABLE IF NOT EXISTS challenge(token varchar, id varchar, secret varchar, provider other, image blob)")
```

Comment:
Serializing the provider to the DB is an innovative idea, but it is very inefficient and flaky.
* Have you checked the size (in bytes) of a stored row?
* What if the provider changes over time (for example, a bug gets fixed).

It would be much better to assign an id to the provider, and store only the id in the database. Assigning a unique id is a little tricky. We can discuss it if you have any doubts.

Reply:
Sure, will look into this problem and get back.